recipes/vtk/all/conanfile.py

```diff
@@ -644,8 +654,17 @@ def _yesno(flag):
 
 
     def build(self):
-        if not self.no_copy_source:     # might be done in def source()
-            apply_conandata_patches(self)
+        apply_conandata_patches(self)
+
+        if self.options.wrap_python and self.settings.build_type == "Debug" and self.settings.os == "Windows":
+            # if you apply the python patch here: https://stackoverflow.com/a/40594968
+            # then you need this hack to compile the python portion of the code in debug
+            python_lib_folder = os.path.realpath(os.path.join(which("python"), "..", "libs")).replace("\\", "/")
+            replace_in_file(os.path.join(self.build_folder,"..", "..", "src", "CMakeLists.txt"),
+                                  "project(VTK)",
+                                  'project(VTK)\nlink_directories("{}")\n'.format(python_lib_folder)
+                                  )
+
```

Comment:
Still necessary.
For some reason, in a lot of places, outside of conan, if you apply the patch here (which I apply in our code base),:
https://stackoverflow.com/a/40594968, it just doesn't find the location of "python310_d.lib" properly. No clue why.

I suspect that very few people actually compile python in debug for day to day use, and so this bug goes unnoticed, but our code base links against python and we do not want to recompile the entirety of the python ecosystem to confirm if it's because of that weird patch or if it's because the upstream cmake is broken. Besides, there's not much gain for us to compile python in debug anyways.

I suspect that this change won't fly with an official cci review, so I'm cool with you just taking it out and I'll apply that fix in my recipe.

Reply:
Hrm.  Seems like your problems will go away with Py >= 3.8.
Is it hard to upgrade Python for you, or for VTK ?

I'm trying to understand... this is for building VTK in debug mode, using Python's release library?  (copied to add _d so the linker finds it, with the header tweak so VTK expects the non-debug API).

Happy to leave in until the review, maybe they have a good solution for this.

A side-note, you can probably use conan.tools.microsoft.unix_path instead of the replace(slashes).